### PR TITLE
on demand items implementation

### PR DESCRIPTION
**Pull request by AlexanderShniperson**
implemented feature #73

### CODE REVIEW

**Comment by sanekyy** [comment]
Hello. 
Can you change names of your new fields and functiono to match the accepted builder approach in flutter : IndexedWidgetBuilder itemBuilder() function to build item on demand instead of onDemandItemContent and itemCount instead of onDemandItemsCount

You can look on examples in ListView.builder() or PageView.builder().

Also please leave the old way as it is with default constructor of CarouselSlider and create CarouselSlider.builder as ListView.builder() or PageView.builder().

**Comment by AlexanderShniperson** [comment]
Hi
Ok thanks, i will do that, take me a time

**Comment by AlexanderShniperson** [comment]
done, please check

File: lib/carousel_slider.dart [hunk 1]
```diff
@@ -22,17 +22,65 @@ class CarouselSlider extends StatefulWidget {
...
+        this.itemCount = items.length,
+        this.itemBuilder = null,
         this.pageController = PageController(
           viewportFraction: viewportFraction,
-          initialPage: enableInfiniteScroll ? realPage + initialPage : initialPage,
-        );
+          initialPage:
+              enableInfiniteScroll ? realPage + initialPage : initialPage,
+        ) {
+    if (onWidgetCreated != null) onWidgetCreated(this);
```
Comment (sanekyy) [inline lib/carousel_slider.dart]: Why did you add this logic?

Comment (AlexanderShniperson) [reply lib/carousel_slider.dart]: i need this callback to take access to 'pageController' then i use them to slide items by ui buttons

Comment (sanekyy) [reply lib/carousel_slider.dart]: It's unnecessary callback cause you always can do this by following simple code:

```
@override
  Widget build(BuildContext context) {
    final carousel = CarouselSlider(...);
    final controller = carousel.pageController;

    return carousel;
  }
```

**Comment by AlexanderShniperson** [comment]
callback removed

**Comment by sanekyy** [comment]
Thx. Hope it'll be merged.

**Comment by AlexanderShniperson** [comment]
Hi, next three days i will be at Hackathon then i will do that, take me a time

**Comment by AlexanderShniperson** [comment]
done, but i can't run sample project because AndroidStudio swears
```
[!] Your app isn't using AndroidX.
    To avoid potential build failures, you can quickly migrate your app by following the steps on https://goo.gl/CP92wY.
```

**Comment by sanekyy** [comment]
Waiting for new version on pub!)

Also, can you please invert order of changelog?

**Comment by serenader2014** [comment]
New version published to the pub, please check it out :)
